sequencer/finalizer.go

```diff
@@ -456,19 +456,23 @@ func (f *finalizer) storeProcessedTx(ctx context.Context, previousL2BlockStateRo
 }
 
 // handleTransactionError handles the error of a transaction
-func (f *finalizer) handleTransactionError(ctx context.Context, result *state.ProcessBatchResponse, tx *TxTracker) {
+func (f *finalizer) handleTransactionError(ctx context.Context, result *state.ProcessBatchResponse, tx *TxTracker) *sync.WaitGroup {
```

Comment:
We are returning the WaitGroup but I don't see any play where we are using the WaitGroup, is it ok?

Reply:
It is for the unit tests in `sequencer/finalizer_test.go` line 1147, to be testable. Otherwise, the next test fails if we don't wait for the queries when we do assert expectations.